theano/tensor/basic.py

```diff
@@ -2312,17 +2315,22 @@ def ones_like(model, dtype=None):
     """
     if dtype is None:
         dtype = model.type.dtype
-    ret = fill(model, constant(1.0, dtype=dtype))
-    return ret
+    ret = constant(1.0, dtype=dtype)
+    if opt and ret.type == model.type:
```

Comment:
I had tried 
1) with this diff, 
2) Returning constant default
3) Returning graph default
Returning graph is the slowest whereas the returning constant is fastest. But I didn't get any error when i tried 2)